automatic_push/ietf_push.py

```diff
@@ -0,0 +1,234 @@
+"""
+Script automatically pushes new IETF RFC and draft yang modules to the GitHub repository:
+https://github.com/yang-catalog/yang. Old ones are removed and their naming is corrected to <name>@<revision>.yang.
+An e-mail with information about local update of new RFCs is sent to yangcatalog admin users if
+there are files to update. Message about new RFC yang modules is also sent to the Cisco Webex Teams,
+room: YANG Catalog Admin.
+"""
+
+__author__ = 'Bohdan Konovalenko'
+__copyright__ = 'Copyright The IETF Trust 2023, All Rights Reserved'
+__license__ = 'Apache License, Version 2.0'
+__email__ = 'bohdan.konovalenko@pantheon.tech'
+
+import filecmp
+import glob
+import json
+import os
+import shutil
+import typing as t
+from configparser import ConfigParser
+from dataclasses import dataclass
+
+import requests
+
+import ietfYangDraftPull.draftPullUtility as dpu
+import utility.log as log
+from automatic_push.utils import get_forked_repository
+from utility import message_factory, repoutil
+from utility.create_config import create_config
+from utility.script_config_dict import script_config_dict
+from utility.scriptConfig import ScriptConfig
+from utility.util import job_log
+
+BASENAME = os.path.basename(__file__)
+FILENAME = BASENAME.split('.py')[0]
+DEFAULT_SCRIPT_CONFIG = ScriptConfig(
+    help=script_config_dict[FILENAME]['help'],
+    args=script_config_dict[FILENAME]['args'],
+    arglist=None if __name__ == '__main__' else [],
+)
+
+
+def main(script_conf: ScriptConfig = DEFAULT_SCRIPT_CONFIG.copy()):
+    args = script_conf.args
+    config = create_config(args.config_path)
+    ietf_push = IetfPush(config, args.send_message)
+    ietf_push()
+
+
+class IetfPush:
+    @dataclass
+    class RFCsCommitResult:
+        commit_successful: bool
+        message: str
+
+    def __init__(self, config: ConfigParser = create_config(), send_message: bool = False):
+        self.send_message = send_message
+
+        self.config = config
+        self.rfc_exceptions_file_path = self.config.get('Directory-Section', 'rfc-exceptions')
+        self.verified_commits_file_path = self.config.get('Directory-Section', 'commit-dir')
+        self.ietf_rfc_url = self.config.get('Web-Section', 'ietf-RFC-tar-private-url')
+        self.ietf_draft_url = config.get('Web-Section', 'ietf-draft-private-url')
+        self.my_uri = config.get('Web-Section', 'my-uri')
+        self.domain_prefix = config.get('Web-Section', 'domain-prefix')
+        ietf_directory = self.config.get('Directory-Section', 'ietf-directory')
+        self.rfc_directory = os.path.join(ietf_directory, 'YANG-rfc')
+        self.is_production = self.config.get('General-Section', 'is-prod') == 'True'
+        log_directory = self.config.get('Directory-Section', 'logs')
+        self.logger = log.get_logger('ietf_push', f'{log_directory}/jobs/ietf-push.log')
+
+        self.repo: t.Optional[repoutil.ModifiableRepoUtil] = None
+
+    @job_log(file_basename=BASENAME)
+    def __call__(self):
```

Comment:
Is there any particular reason for using this approach with using `__call__` method?
I am not saying it is incorrect, it is not used anywhere in the solution.

Reply:
this class is only responsible for one job, so I thought it is reasonable to use this method here (and for IANA push as well)